backend/service.bal

```diff
@@ -130,4 +130,94 @@ service http:InterceptableService / on new http:Listener(9090) {
 
         return result;
     }
+
+    # Update the favorite status of an application for the authenticated user.
+    #
+    # + ctx - Request context containing authenticated user information and JWT payload
+    # + id - The unique identifier of the application to update (must be a valid integer)
+    # + active - The favorite status to set: "1" to add to favorites, "0" to remove from favorites
+    # + return - Success response with confirmation message, or error responses for various failure scenarios
+    resource function patch apps(http:RequestContext ctx, string id, string active)
+        returns http:Ok|http:NotFound|http:BadRequest|http:InternalServerError|http:NotModified {
+
+        authorization:CustomJwtPayload|error userInfo = ctx.getWithType(authorization:HEADER_USER_INFO);
+        if userInfo is error {
+            log:printError(USER_INFO_HEADER_NOT_FOUND_ERROR, userInfo);
+            return <http:InternalServerError>{
+                body: {message: "Authentication information not found in request. Please ensure you are logged in."}
+            };
+        }
+
+        int|error appId = int:fromString(id);
+        int|error isFav = int:fromString(active);
```

Comment:
Can't we handle this from the frontend? We can send int type variables to the endpoint